packages/terra-flex-box/src/FlexBox.jsx

```diff
@@ -0,0 +1,155 @@
+  /**
+   * This is a shorthand flex-direction and flex-wrap properties, which together define the flex container's main and cross axes.
+   */
+  flexFlow: PropTypes.string,
+  /**
+   * Defines the ability for a flex item to grow if necessary.
+   */
+  flexGrow: PropTypes.number,
+  /**
+   * Defines the ability for a flex item to shrink if necessary.
+   */
+  flexShrink: PropTypes.number,
+  /**
+   * Allow the items to wrap as needed.
+   */
+  flexWrap: PropTypes.oneOf(['nowrap', 'wrap-reverse', 'wrap']),
+  /**
+   * Defines the height of the container.
+   */
+  height: PropTypes.string,
+  /**
+   * Defines the alignment along the main axis.
+   */
+  justifyContent: PropTypes.oneOf(justifyContentTypes),
+  /**
+   * Defines the maximum height for the container.
+   */
+  maxHeight: PropTypes.string,
+  /**
+   * Defines the maximum width for the container.
+   */
+  maxWidth: PropTypes.string,
+  /**
+   * Defines the minimum height for the container.
+   */
+  minHeight: PropTypes.string,
+  /**
+   * Defines the minimum width for the container.
+   */
+  minWidth: PropTypes.string,
+  /**
+   * Controls the order in which items appear in the container.
+   */
+  order: PropTypes.number,
+  /**
+   * Defines the width for the container.
+   */
+  width: PropTypes.string,
+};
+
+const defaultProps = {
+  display: 'flex',
+  element: 'div',
+};
+
+const FlexBox = ({
+  alignContent,
+  alignItems,
+  alignSelf,
+  children,
+  display,
+  element,
+  flex,
+  flexBasis,
+  flexDirection,
+  flexFlow,
+  flexGrow,
+  flexShrink,
+  flexWrap,
+  height,
+  justifyContent,
+  maxHeight,
+  maxWidth,
+  minHeight,
+  minWidth,
+  order,
+  width,
+  ...props
+}) => {
+  const customProps = props;
+  if (!customProps.style) { customProps.style = {}; }
+  if (alignContent) { customProps.style.alignContent = alignContent; }
+  if (alignItems) { customProps.style.alignItems = alignItems; }
+  if (alignSelf) { customProps.style.alignSelf = alignSelf; }
+  if (display) { customProps.style.display = display; }
+  if (flex) { customProps.style.flex = flex; }
+  if (flexBasis) { customProps.style.flexBasis = flexBasis; }
+  if (flexDirection) { customProps.style.flexDirection = flexDirection; }
+  if (flexFlow) { customProps.style.flexFlow = flexFlow; }
+  if (flexGrow !== undefined) { customProps.style.flexGrow = flexGrow; }
+  if (flexShrink !== undefined) { customProps.style.flexShrink = flexShrink; }
+  if (flexWrap) { customProps.style.flexWrap = flexWrap; }
+  if (height) { customProps.style.height = height; }
+  if (justifyContent) { customProps.style.justifyContent = justifyContent; }
+  if (maxHeight) { customProps.style.maxHeight = maxHeight; }
+  if (maxWidth) { customProps.style.maxWidth = maxWidth; }
+  if (minHeight) { customProps.style.minHeight = minHeight; }
+  if (minWidth) { customProps.style.minWidth = minWidth; }
+  if (order !== undefined) { customProps.style.order = order; }
+  if (width) { customProps.style.width = width; }
```

Comment:
FWIW this can all be cleaned up if you use [aphrodite](https://github.com/Khan/aphrodite). Its the library that generates the CSS for dynamic grid.

Reply:
updated to use aphrodite and cleaned up the code a bit more